### ISSUE TITLE
v0.9.0 can't find libdevmapper.so.1.02

### ISSUE THREAD

**Issue by maresb** [wagoodman/dive#242]
For stability and compatibility, I run Ubuntu 18.04.3 LTS.  I just installed `dive` v0.9.0, and when I run `dive` I get the error
```
dive: error while loading shared libraries: libdevmapper.so.1.02: cannot open shared object file: No such file or directory
```
I thought I might be able to fix it with `sudo ln -s /lib/x86_64-linux-gnu/libdevmapper.so.1.02.1 /lib/x86_64-linux-gnu/libdevmapper.so.1.02`, but then I get the error
```
dive: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by dive)
```
My `libc6` version is `2.27-3ubuntu1`.  Since I'm not aware of a simple way to upgrade `libc6` under Ubuntu, I did not attempt to further debug the issue.

After downgrading to v0.8.1, `dive` just works.

**Comment by 1138-4EB** [comment]
@maresb, as commented in #243, until `v0.8.1`, `dive` was statically compiled. However, some dependencies that were introduced in `v0.9.0` require it not to be statically linked anymore. As a result, `dive` is now dependent on the version of the standard library that is available on the host where it is built. Possible solutions are:

- Use docker images instead of installing `dive` on your host.
- Install golang and build dependencies, and compile `dive` on your host.
- Use an `ubuntu:18.04` container to install dependencies and compile `dive`. Then, install the binary on your host.
- Add a GitHub Actions workflow (or any other CI job) to this repo in order to build `dive` for different distros (using Docker containers) and generate multiple `deb`/`rpm` packages.
- Add `deb` or `rpm` recipes to the official repositories of Debian/Ubuntu/Fedora/Arch/Alpine so that the binary is built for each version.

**Comment by maresb** [comment]
Thanks @1138-4EB for the explanation!  In any case, the documentation at https://github.com/wagoodman/dive#installation is broken for Ubuntu 18.04.  In the meantime, perhaps the *Docker* subsection should be moved to the top and indicated as "recommended".

**Comment by Sarke** [comment]
Same issue.

I installed using the deb file from releases.  Should the dependencies not be listed in the deb file and installed automatically?

**Comment by 1138-4EB** [comment]
@Sarke, currently `deb` and `rpm` packages generated release by `dive` are just tarballs with a different format. I.e., there is no metadata. While it would be possible to declare dependencies, such as `gpgme device-mapper`, that would not solve the issue about the dependency of GLIBC. Although possible, installing multiple versions of GLIBC is probably looking for problems.

**Comment by renanbr** [comment]
using docker as @1138-4EB suggested works for me, adding

**Comment by Sarke** [comment]
Thanks.  I think we can tweak that a bit, since I use config file for dive we need to mount that too.

**Comment by wagoodman** [comment]
Indeed, both the deb and rpm were quite broken. This morning I released 0.9.1 which should fix these issues by removing the offending dependencies (from libpod) and going back to a statically compiled binary (while keeping Podman support too!). @1138-4EB / @renanbr thanks for posting a good workaround. 

Let me know if you have any problems with 0.9.1!

**Comment by 1138-4EB** [comment]
@wagoodman, precisely, I didn't say before that breaking static builds was/is a bad idea, because this is your project and you are entitled to add any built-in dependencies that you wish. You seemed to be quite excited about supporting podman, and I am no one to despise that. Nonetheless, I thought it was not a good move, precisely because generating debs and rpms for multiple versions of Debian/Ubuntu/Fedora/Suse is an absolute nightmare for such an small project. Therefore, I'm very happy and grateful with you deciding to handle podman differently. This is specially so because podman is not supported on all the platforms yet.

However, I think that @Sarke's comment deserves some attention. deb and rpm files should contain not only the binary, but also other metadata. For rpms, https://copr.fedorainfracloud.org/ allows to test RHEL, CentOS, Fedora and Suse. I don't know if such a service exists for deb packages.
